docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md

```diff
@@ -806,6 +806,167 @@ One challenge when developing locally is that a developer might make modificatio
+    )
+    // This is getting called for every entry node so we don't want the console logs to get cluttered
+    warnOnceForNoSupport = true
+    // highlight-end
+  }
+}
+```
+
+#### Content Sync data source (or CMS) changes
+
+The CMS will need to send a webhook when content is changed
+
+##### Using the gatsby preview extension
+
+This package is currently in the process of being open sourced and should be available shortly. For context, here is what the package provides, a button to handle opening the Content Sync waiting room. You will need to create a button similar to this for the time being.
+![Open Preview Button Screenshot](https://user-images.githubusercontent.com/18426780/61498855-37e35a00-a982-11e9-8201-ab96be74f1f1.png)
+
+##### Configuration
+
+You will need to store the Content Sync URL from a given Gatsby Cloud site. This will look something like `https://gatsbyjs.com/content-sync/<siteId>`. This is often done in the plugin extension or app extension configuration, this will differ from CMS to CMS depending on how they handle their plugin ecosystem.
+
+##### Building up the manifest id
+
+Recall that we need to create a matching manifest id in the CMS AND the Gatsby plugin. Whenever content is saved we can build up a new manifest id that will look the same as the manifest id we created in the source plugin
+
+```javascript
+const manifestId = `${project.id}-${entryItem.id}-${updatedAt}`
+```
+
+In the CMS we should have access to the project id (if the CMS uses one) and the content id, the only thing we need to figure out how to grab is the timestamp that the content was updated at.
+
+##### Opening the Content Sync waiting room
+
+When we've built up the manifest id, when the "Open Preview" button is clicked, we need to open a new window to the Content Sync waiting room as follows `https://gatsbyjs.com/content-sync/<siteId>/<sourcePluginName>/<manifestId>`. This will ensure that this piece of content is linked to the corresponding content that was created in their Gatsby site.
+
+#### Useful Content Sync development tips
+
+There are a couple gotchas depending on how the CMS acts. Sometimes you will need to wait to make sure you have the correct `updatedAt` timestamp as some CMS may take a second to update their backend and then wait for the change to propagate to the frontend. While others will immediately update the frontend and then propagate that to the backend.
+
+Make sure that a preview webhook is being sent to Gatsby Cloud after the content is edited, whether it's before you press the "Open Preview" button or the "Open Preview" is the trigger that sends the webhook.
+
+While developing, you can set the Gatsby `VERBOSE` env variable to `"true"` to see additional logs that will help you debug what's happening in the source plugin. Ideally when you click the `Open Preview` button in the CMS the `manifestId` in the URL should match the `manifestId` that the source plugin creates from that revision.
```

Comment:
Breaking up all of the unique items into separate bullets for easier scanning and more clear organization...


```suggestion
- While developing, you can set the Gatsby `VERBOSE` env variable to `"true"` to see additional logs that will help you debug what's happening in the source plugin.
- Ideally when you click the `Open Preview` button in the CMS the `manifestId` in the URL should match the `manifestId` that the source plugin creates from that revision.
```

Reply:
I'm a little confused what you mean by this? Doesn't this 100% need to be the case in order for Content Sync to work?

> Ideally when you click the `Open Preview` button in the CMS the `manifestId` in the URL should match the `manifestId` that the source plugin creates from that revision.

Reply:
It's the happy path 😆 , yeah I'll change that to make it sound more like that's the only way it works

Reply:
Okay sweet, thanks for clarifying! (and making the tweak) :smile: